legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabPatchApi.java

```diff
@@ -0,0 +1,357 @@
+        {
+            targetVersion = getProjectVersion(projectId, patchReleaseVersionId.getMajorVersion(), patchReleaseVersionId.getMinorVersion(), patchReleaseVersionId.getPatchVersion());
+            throw new LegendSDLCServerException(patchReleaseVersionId.toVersionIdString() + " already exists", Response.Status.BAD_REQUEST);
+        }
+        catch (Exception e)
+        {
+            LOGGER.warn("Error in fetching release tag for version ", patchReleaseVersionId, projectId, e);
+        }
+        if (targetVersion != null)
+        {
+            throw new LegendSDLCServerException(patchReleaseVersionId.toVersionIdString() + " already exists", Response.Status.BAD_REQUEST);
+        }
+
+        // Check if the patch branch we want to release exists or not
+        Branch patchBranch = null;
+        try
+        {
+            patchBranch = getGitLabApi().getRepositoryApi().getBranch(gitLabProjectId.getGitLabId(), getPatchReleaseBranchName(patchReleaseVersionId));
+        }
+        catch (Exception e)
+        {
+            LOGGER.warn("Error in fetching the patch release branch ", patchReleaseVersionId, projectId, e);
+        }
+        if (patchBranch == null)
+        {
+            throw new LegendSDLCServerException("Patch release branch for " + patchReleaseVersionId + " doesn not exist for the project " + projectId, Response.Status.BAD_REQUEST);
+        }
+        Version releaseVersion =  newVersion(gitLabProjectId, patchReleaseVersionId, patchBranch.getCommit().getId(), VersionId.newVersionId(patchReleaseVersionId.getMajorVersion(), patchReleaseVersionId.getMinorVersion(), patchReleaseVersionId.getPatchVersion()), "");
+
+        // delete the patch release branch and close the MRs created for this branch
+        submitBackgroundRetryableTask(() -> GitLabApiTools.deleteBranchAndVerify(getGitLabApi().getRepositoryApi(), gitLabProjectId.getGitLabId(), getPatchReleaseBranchName(patchReleaseVersionId), 20, 1_000), 5000L, "delete " + getPatchReleaseBranchName(patchReleaseVersionId));
+        submitBackgroundRetryableTask(() -> closeMergeRequestsCreatedForPatchReleaseBranch(gitLabProjectId, patchReleaseVersionId), 5000L, "close merge requests created for branch " + getPatchReleaseBranchName(patchReleaseVersionId));
+        return releaseVersion;
+    }
+
+    private boolean closeMergeRequestsCreatedForPatchReleaseBranch(GitLabProjectId projectId, VersionId patchReleaseVersionId)
+    {
+        List<MergeRequest> mergeRequests = Collections.emptyList();
+        String branchName = getSourceBranch(projectId, patchReleaseVersionId);
+        try
+        {
+           mergeRequests = getGitLabApi().getMergeRequestApi().getMergeRequests(projectId.getGitLabId()).stream().filter(mr -> branchName.equals(mr.getTargetBranch())).collect(Collectors.toList());
+        }
+        catch (Exception e)
+        {
+            LOGGER.warn("Unable to fetch merge requests for the project " + projectId.getGitLabId(), e);
+        }
```

Comment:
You should create a MergeRequestFilter that filters on the target branch (and only gets open MRs). That way the MRs will be filtered on the GitLab side. And it's better to use a Pager instead of instantiating the full list.

Also, if there's an error getting the MRs that suggests a retry might work (see, e.g., GitLabApiTools.isRetryableGitLabApiException), then the method should return false so that the task will be put back on the queue to try again.